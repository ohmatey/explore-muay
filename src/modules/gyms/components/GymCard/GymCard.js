import React from 'react'
import propTypes from 'prop-types'

const GymCard = ({
  title,
  subtitle,
  handleClick
}) => (
  <div>
    <h1>{title}</h1>
    <h2>{subtitle}</h2>

    <button onClick={handleClick}>Go</button>
  </div>
)

GymCard.propTypes = {
  title: propTypes.string
}

export default GymCard
