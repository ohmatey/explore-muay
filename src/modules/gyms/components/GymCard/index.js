import GymCard from './GymCard'

export default GymCard
