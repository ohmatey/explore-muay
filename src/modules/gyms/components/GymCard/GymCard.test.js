import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import GymCard from './GymCard'

describe('<GymCard />', () => {
  let component

  beforeEach(() => {
    component = shallow(<GymCard />)
  })

  it('should render with no errors', () => {
    const tree = toJson(component)

    expect(tree).toMatchSnapshot()
  })

  it('it should render with header and subheader', () => {
    const props = {
      title: 'Bobs Muay Thai',
      subtitle: 'Bob Brown'
    }

    component.setProps(props)

    expect(component.find('h1').length).toEqual(1)
    expect(component.find('h1').text()).toEqual(props.title)

    expect(component.find('h2').length).toEqual(1)
    expect(component.find('h2').text()).toEqual(props.subtitle)

    expect(component.find('button').length).toEqual(1)
  })

  it('should call function when button is clicked', () => {
    const mockHandleClick = jest.fn()

    component.setProps({
      handleClick: mockHandleClick
    })

    component.find('button').simulate('click')

    expect(mockHandleClick).toHaveBeenCalled()
  })
})
