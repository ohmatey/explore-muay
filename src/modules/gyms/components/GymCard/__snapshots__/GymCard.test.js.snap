// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GymCard /> should render with no errors 1`] = `
<div>
  <h1 />
  <h2 />
  <button>
    Go
  </button>
</div>
`;
